src/pieces/pawn.js

```diff
@@ -1,23 +1,85 @@
 import Piece from './piece';
-
+import Board from '../board';
 class Pawn extends Piece {
   constructor(row, column, side) {
     super(row, column, side);
     this.name = 'pawn';
     this.display = `<i class="fas fa-chess-pawn ${side}"></i>`;
   }
 
-  findLegalMoves() {
+  findLegalMoves(board) {
     const possibleMoves = [];
-    
-    if (this.side == 'white') {
-      if (this.row - 1 >= 0) possibleMoves.push([this.row - 1, this.column]);
-      if (this.row - 2 >= 0 && this.row == 6) possibleMoves.push([this.row - 2, this.column]);
+
+    if (this.side === 'white') {
+      for (let i = 1; i <= 1; i++) {
+        if (this.row - 1 >= 0) {
```

Comment:
jeżeli nic nie pominąłem ten `if` to jest jedyna rzecz jaka się różni między białymi i czarnymi. Proponuje wynieść tą logikę na samą góre policzyć zmienną `nextRow` w oparciu `this.side` po to by całą reszte logiki zapisać potem już tylko raz